Sources/App/Controllers/PackageController+routes.swift

```diff
@@ -18,20 +18,33 @@ import Vapor
 
 struct PackageController {
     
-    func show(req: Request) throws -> EventLoopFuture<HTML> {
+    func show(req: Request) async throws -> HTML {
         guard
             let owner = req.parameters.get("owner"),
             let repository = req.parameters.get("repository")
         else {
-            return req.eventLoop.future(error: Abort(.notFound))
+            throw Abort(.notFound)
+        }
+
+        if repository.lowercased().hasSuffix(".git") {
+            throw Abort.redirect(to: SiteURL.package(.value(owner),
+                                                     .value(repository.droppingGitExtension),
+                                                     .none).absoluteURL(),
+                                 type: .permanent)
+        }
+
+        do {
+            return try await ShowRoute
+                .query(on: req.db, owner: owner, repository: repository)
+                .map {
+                    PackageShow.View(path: req.url.path,
+                                     model: $0.model, packageSchema: $0.schema)
+                        .document()
+                }.get()
+        } catch let error as AbortError where error.status == .notFound {
+            throw Abort.redirect(to: "https://github.com/\(owner)/\(repository)",
+                                 type: .permanent)
```

Comment:
Rather than blindly re-direct to Github here we could (should) check if it exists first.

Reply:
I'll review this properly later, but while I notice this specifically, we want to be very careful with permanent redirects. this tells Google that this URL is not and never will be valid. This should be a temporary redierct.

Reply:
But, I'm not sure a server-side redirect is the best choice here as this might be a good opportunity to ask people to add the package, accompanied with a link to the GitHub page.

We need to think about this a bit before merging this one. I'm writing today, but I'll have a think about it over the weekend.

Reply:
Yeah, this wasn't for merging, it was for discussion!